Contents/Code/siteFuckingAwesome.py

```diff
@@ -0,0 +1,128 @@
+import PAsearchSites
+import PAgenres
+import PAactors
+
+def search(results,encodedTitle,title,searchTitle,siteNum,lang,searchByDateActor,searchDate,searchSiteID):
+    if searchSiteID != 9999:
+        siteNum = searchSiteID
+    searchResults = HTML.ElementFromURL(PAsearchSites.getSearchSearchURL(siteNum) + encodedTitle)
+    for searchResult in searchResults.xpath('//div[@class="gallery"]/div'):
+        titleNoFormatting = searchResult.xpath('.//div[@class="video-title truncate"]/a')[0].text_content().strip()
+        Log("Title: " + titleNoFormatting)
+        curID =('https://fuckingawesome.com' + searchResult.xpath('.//div[@class="video-title truncate"]/a')[0].get('href')).replace('/','_').replace('?','!')
+        Log("curID: " + curID)
+        releaseDate = parse(searchResult.xpath('.//span[@class="small date"]')[0].text_content().strip()).strftime('%Y-%m-%d')
+        Log("releaseDate: " + releaseDate)
+        firstActor = searchResult.xpath('.//span[@class="subtitle small"]/a')[0].text_content().strip()
+        Log("firstActor: " + firstActor)
+        if searchDate:
+            score = 100 - Util.LevenshteinDistance(searchDate, releaseDate)
+        else:
+            score = 100 - Util.LevenshteinDistance(searchTitle.lower(), titleNoFormatting.lower())
+        
+        results.Append(MetadataSearchResult(id = curID + "|" + str(siteNum), name = releaseDate + " " + firstActor + " in " + titleNoFormatting + " [FuckingAwesome.com] ", score = score, lang = lang))
+
+    return results
+
+def update(metadata,siteID,movieGenres,movieActors):
+    Log('******UPDATE CALLED*******')
+
+    url = str(metadata.id).split("|")[0].replace('_','/').replace('?','!')
+    detailsPageElements = HTML.ElementFromURL(url)
+    art = []
+    metadata.collections.clear()
+    movieGenres.clearGenres()
+    movieActors.clearActors()
+
+    # Studio
+    metadata.studio = 'FuckingAwesome'
+
+    # Title
+    metadata.title = detailsPageElements.xpath('//h1')[0].text_content().strip()
+
+    # Summary
+    metadata.summary = detailsPageElements.xpath('//div[@class="more text-justify"]')[0].text_content().strip()
+
+    #Tagline and Collection(s)
+    tagline = PAsearchSites.getSearchSiteName(siteID).strip()
+    metadata.tagline = tagline
+    metadata.collections.add(tagline)
+
+    # Genres
+    genres = detailsPageElements.xpath('//div[@class="tags"]/ul/li/a')
+    if len(genres) > 0:
+        for genreLink in genres:
+            genreName = genreLink.text_content().strip().lower()
+            movieGenres.addGenre(genreName)
+    movieGenres.addGenre("Genre")
```

Comment:
Feel free to remove this line, as it just adds the genre "Genre".

Reply:
ahaaa i forgett it every time